cmd/edit/machinepool/cmd.go

```diff
@@ -245,8 +273,76 @@ func run(cmd *cobra.Command, argv []string) {
 		os.Exit(1)
 	}
 
+	labels := args.labels
+	labelMap := make(map[string]string)
+	if interactive.Enabled() {
+		if labels == "" {
+			for lk, lv := range machinePool.Labels() {
+				if labels != "" {
+					labels += ","
+				}
+				labels += fmt.Sprintf("%s=%s", lk, lv)
+			}
+		}
+		labels, err = interactive.GetString(interactive.Input{
+			Question: "Labels",
+			Help:     cmd.Flags().Lookup("labels").Usage,
+			Default:  labels,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid comma-separated list of attributes: %s", err)
+			os.Exit(1)
+		}
+	}
+	labels = strings.Trim(labels, " ")
+	if labels != "" {
+		for _, label := range strings.Split(labels, ",") {
+			if !strings.Contains(label, "=") {
+				reporter.Errorf("Expected key=value format for labels")
+				os.Exit(1)
+			}
+			tokens := strings.Split(label, "=")
+			labelMap[strings.TrimSpace(tokens[0])] = strings.TrimSpace(tokens[1])
+		}
+	}
+
+	taints := args.taints
+	taintBuilders := []*cmv1.TaintBuilder{}
+	if interactive.Enabled() {
+		if taints == "" {
+			for _, taint := range machinePool.Taints() {
+				if taints != "" {
+					taints += ","
+				}
+				taints += fmt.Sprintf("%s=%s:%s", taint.Key(), taint.Value(), taint.Effect())
+			}
+		}
+		taints, err = interactive.GetString(interactive.Input{
+			Question: "Taints",
+			Help:     cmd.Flags().Lookup("taints").Usage,
+			Default:  taints,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid comma-separated list of attributes: %s", err)
+			os.Exit(1)
+		}
+	}
+	taints = strings.Trim(taints, " ")
+	if taints != "" {
+		for _, taint := range strings.Split(taints, ",") {
+			if !strings.Contains(taint, "=") || !strings.Contains(taint, ":") {
+				reporter.Errorf("Expected key=value:scheduleType format for taints")
+				os.Exit(1)
+			}
+			tokens := strings.FieldsFunc(taint, Split)
+			taintBuilders = append(taintBuilders, cmv1.NewTaint().Key(tokens[0]).Value(tokens[1]).Effect(tokens[2]))
+		}
+	}
+
 	mpBuilder := cmv1.NewMachinePool().
-		ID(machinePool.ID())
+		ID(machinePool.ID()).
+		Labels(labelMap).
+		Taints(taintBuilders...)
```

Comment:
So labels and taints are always sent? 
How does the API handle the difference between no change and reset to empty?

Reply:
Both are sent to the backend.

Reply:
I changed how this is done (which addresses all the comments above). If the user explicitly passes the flag (or if the interactive mode is used, since that always displays labels and taints), then we send it to the API regardless of whether there's a value or not. That way, if the user sends `--labels ""` (or enters an empty string  in interactive mode) it clears the labels.